File: commons/javafx/src/main/java/com/cedarsoft/commons/javafx/IntegerHexTextFormatterFilter.java
package com.cedarsoft.commons.javafx;

import java.util.function.Predicate;

import javax.annotation.Nonnull;

/**
 * Processes text formatter changes. Ensures only valid integers are added
 */
class IntegerHexTextFormatterFilter extends NumberTextFormatterFilter {

  IntegerHexTextFormatterFilter(@Nonnull NumberStringConverterForIntegerHex converter, @Nonnull Predicate<Integer> filter) {
    super(converter, number -> filter.test(number.intValue()), 0);
  }

}

File: commons/javafx/src/main/kotlin/it/neckar/open/javafx/IntegerHexTextFormatterFilter.kt
package it.neckar.open.javafx

import java.util.function.Predicate

/**
 * Processes text formatter changes. Ensures only valid integers are added
 */
internal class IntegerHexTextFormatterFilter(
  converter: NumberStringConverterForIntegerHex,
  filter: Predicate<Int>,
) : NumberTextFormatterFilter(
  converter = converter,
  filter = Predicate { number: Number -> filter.test(number.toInt()) },
  emptyValue = 0
)
